app/server.py

```diff
@@ -31,37 +32,36 @@ def bad_request(err):
 
 @app.route('/api/v1/search/<search_engine>', methods=['GET'])
 def search(search_engine):
+    search_engine = search_engine.strip().lower()
     try:
         num = request.args.get('num') or 10
         count = int(num)
-        qformat = request.args.get('format') or 'json'
+        qformat = request.args.get('format', 'json')
         if qformat not in ('json', 'xml'):
             abort(400, 'Not Found - undefined format')
 
-        engine = search_engine
-        if engine not in ('google', 'bing', 'duckduckgo', 'yahoo', 'ask',
```

Comment:
Check the other comment, I think this should stay.

Reply:
The reason that I got rid the list is so that we do not need to modify this file each time that we add a new engine.  Under this do not repeat (DRY) approach, the process for adding a new engine is to define a new subclass of Scrapers and add it to the dictionary.  Under the current approach, if the list in one file gets out of sync with the dictionary in the other file then hard to spot bugs can result.

Reply:
This was earlier done so that if some get request comes from other application than we could check if the search engine is supported or not.So it should retain.
Consider: if some one makes a request for https://query-server.herokuapp.com/api/v1/search/gigablast
So how will you deal with it , if you remove this ??

Reply:
What will happen is explained in the comment just above this one and in the code and comment on lines 49 and 50 just below.  [gigablast is not in the scrapers dictionary.](https://github.com/fossasia/query-server/pull/280/files#diff-ae74241666286cc647595b08d2de8ac2R11)  This means that line 49 (just below) will raise a KeyError and line 51 will tell the user that gigablast is an unsupported search engine.  The comment on line 50 makes this intention clear.  The reason for not repeating the names of all scrapers in two different places in two different files is https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

Reply:
Now, I am sure that the code this PR has not been tested. :disappointed:

Reply:
Under this PR, those calling apps would also get the same error.  Neither ‘gigablast’ nor ‘g’ would be in the dictionary so they would receive a raised exception instead of being mistakenly handed the Google scraper in a way similar to issue #271 

A dictionary whose key is the whole name of the scraper is safer than a dictionary whose key is merely the first name of the scraper.  Do you get it or not?

Reply:
Repeating the names of the scrapers in a list in one file and a dict in another file goes against DRY and can be a source of hard to find bugs.